Add new builtin plugin for msitools binaries

JonoYang opened the pull request:
This is a new builtin plugin that compiles msitools and packs the binaries into a locationprovider plugin. This is used for https://github.com/nexB/scancode-toolkit/issues/2559 and related to https://github.com/nexB/scancode-toolkit/issues/2565

pombredanne commented on builtins/packagedcode_msitools-linux/Makefile:
```diff
@@ -0,0 +1,33 @@
+# SPDX-License-Identifier: Apache-2.0
```
We now use  a simplified header. Can you check in nearby plugins for inspiration? (I wish we could use the skeleton to merge in subdirs... may be this is possible too? )

JonoYang replied:
I've updated the file header.

pombredanne commented on builtins/packagedcode_msitools-linux/fetchcode_container.ABOUT:
```diff
@@ -0,0 +1,12 @@
+about_resource: .
```
You probably meant to use this as a base for msitools?

JonoYang replied:
Yes. I've created the file `packagedcode_msitools.ABOUT` and removed this file.

pombredanne commented on builtins/packagedcode_msitools-linux/lib-src/bats-core-ce5ca2802fabe5dc38393240cd40e20f8928d3b0.zip.ABOUT:
```diff
@@ -0,0 +1,9 @@
+about_resource: bats-core-ce5ca2802fabe5dc38393240cd40e20f8928d3b0.zip
```
Are you sure this is needed just to build msitools?

JonoYang replied:
The meson buildscript requires this. A git submodule of bats-core is used during build time: https://gitlab.gnome.org/GNOME/msitools/-/tree/master/subprojects

pombredanne reviewed (changes requested):
Thanks . See a few comments inline

JonoYang commented:
@pombredanne I've addressed your comments. Quick question about license: I noticed that `builtins/packagedcode_msitools-linux/src/packagedcode_msitools/bin/msidiff` and `builtins/packagedcode_msitools-linux/src/packagedcode_msitools/bin/msidump` have the gpl-2.0-plus license header. The library `msitools` is licensed under lgpl-2.1-plus. Would the proper license expression for `packagedcode_msitools` be `apache-2.0 AND lgpl-2.1-plus AND gpl-2.0-plus`?

pombredanne commented on builtins/packagedcode_msitools-linux/Makefile:
```diff
@@ -0,0 +1,19 @@
+	rm -f src/packagedcode_msitools/bin/msi*
+	python setup.py clean --all sdist
+	rm -f src/packagedcode_msitools/bin/msi*
+	GOOS=linux GOARCH=amd64 python setup.py clean --all bdist_wheel --python-tag py3 --plat-name manylinux1_x86_64
```
GOOS=linux GOARCH=amd64  are Goi-specific abd likely not needed.

pombredanne commented on builtins/packagedcode_msitools-linux/setup.cfg:
```diff
@@ -0,0 +1,10 @@
+license_files =
+    apache-2.0.LICENSE
+    gpl-2.0.LICENSE
+    lgpl-2.0.LICENSE
```
this may not be aligned with the actual licenses?

pombredanne commented on builtins/packagedcode_msitools-linux/lib-src/mit.LICENSE:
```diff
@@ -0,0 +1,18 @@
+Permission is hereby granted, free of charge, to any person obtaining
```
You may want to add this to the expression

pombredanne commented on builtins/packagedcode_msitools-linux/setup.py:
```diff
@@ -0,0 +1,69 @@
+
+setup(
+    name="packagedcode-msitools",
+    version="0.101",
```
add some trailing some date as an extra version segment (see other plugins)

pombredanne commented on builtins/packagedcode_msitools-linux/setup.py:
```diff
@@ -0,0 +1,69 @@
+setup(
+    name="packagedcode-msitools",
+    version="0.101",
+    license="lgpl-2.1-plus",
```
Update this as discussed: GPL is needed too

pombredanne commented on builtins/packagedcode_msitools-linux/src/packagedcode_msitools/__init__.py:
```diff
@@ -0,0 +1,29 @@
+            'packagedcode_msitools.msidiff': join(bin_dir, 'msidiff'),
+            'packagedcode_msitools.msidump': join(bin_dir, 'msidump'),
+            'packagedcode_msitools.msiextract': join(bin_dir, 'msiextract'),
+            'packagedcode_msitools.msiinfo': join(bin_dir, 'msiinfo'),
```
Let's use only the ones we need.

pombredanne reviewed (approved):
See a few nit inline. LGTM otherwise
